src/common/portal-metadata-context/PortalMetadataContext.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { IContextItem, IPortalMetadata } from "./Interfaces";
+
+export class PortalMetadataContext {
+
+    private portalMetadata: IPortalMetadata;
+    private portalMetadataContext: IContextItem;
+
+
+    public getPortalMetadataContext() {
+        // returns portalMetadataContext
+    }
+
+    public onFileUpdate() {
+        // updates metadatacontext when file updates
+    }
+
+    private updatePortalMetadataContext() {
+        // updates portalMetadataContext
+    }
+
+    private updatePortalMetadata() {
```

Comment:
we need get/set for portalMetadata too in addtion to update. Moreover update should target at more granular updates and not entire list reset